### PR TITLE
Clarify client's access to artifact store

### PR DESCRIPTION
**Pull request by fabboe**
## What changes are proposed in this pull request?

Doc update to clarify that client needs direct access to artifact store.

Hopefully addresses #2044 #2045 

### What component(s) does this PR affect?

- [ ] UI
- [ ] CLI
- [ ] API
- [ ] REST-API
- [ ] Examples
- [X] Docs
- [ ] Tracking
- [ ] Projects
- [ ] Artifacts
- [ ] Models
- [ ] Scoring
- [ ] Serving
- [ ] R
- [ ] Java
- [ ] Python

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [x] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: docs/source/tracking.rst [hunk 1]
```diff
@@ -454,6 +456,8 @@ For example, if you have a Minio server at 1.2.3.4 on port 9000:
 .. code-block:: bash
 
   export MLFLOW_S3_ENDPOINT_URL=http://1.2.3.4:9000
+  
+on both, tracking server and client.
```
Comment (mparkhe) [inline docs/source/tracking.rst]: Why is this change required? All logging APIs will operate at run level and download APIs will return URIs

**Review by mparkhe — COMMENTED** [review]
Some recommended changes. 

cc: @stbof

File: docs/source/tracking.rst [hunk 2]
```diff
@@ -411,13 +411,15 @@ default location to store artifacts for all runs in this experiment. Additional,
 is a property on :py:class:`mlflow.entities.RunInfo` to indicate location where all artifacts for
 this run are stored.
 
-Use ``--default-artifact-root`` (defaults to local ``./mlruns`` directory) to configure default
-location to server's artifact store. This will be used as artifact location for newly-created
+Use ``--default-artifact-root`` (defaults to client's local ``./mlruns`` directory) to configure the default
```
[the remark this answers is not in the document]
Comment (fabboe) [reply docs/source/tracking.rst]: IMHO artifact paths are always interpreted from the client. E.g. I observe that, when I run `mlflow server` on a remote machine in e.g. `/mnt/bla` that my client tries to write artifacts to its local `/mnt/bla/mlruns/` which it probably can't because that path doesn't exist. We can rephrase to
```
defaults to server's local ``./mlruns`` path which will be used from the client
``` 
?

File: docs/source/tracking.rst (hunk 1 above)
Comment (fabboe) [reply docs/source/tracking.rst]: I think this mainly affects `log_artifact` - not sure I understand your comment. 

Since this is the `Server` section I wanted to emphasize that also the client needs this config, even if it has the tracking server (which itself knows the custom endpoint) configured.
